index.rst

```diff
@@ -1,42 +1,621 @@
+
+.. math::
+  \phi_\mathrm{dir}({\bf r},{\bf s}) =
+    \sum_{i \in R({\bf r})} w_i({\bf r}) \, \phi_i({\bf r}, {\bf s})
+  :label: eq:coaddpsf
+
+.. math::
+  C_\mathrm{dir}({\bf r}, {\bf s}) =
+    \sum_{i \in \left[R({\bf r}) \cap R({\bf s})\right]} \!\!\!
+        w_i({\bf r}) \, w_i({\bf s})
+        \, C_i({\bf r}, {\bf s})
+
+These quantities would be many times larger than the coadd itself if evaluated on every pixel, making direct evaluation impractical.  They are discontinuous at the boundaries of input images (and masked regions within them), making interpolation problematic as well.
+
+The solution we have adopted for PSF models has been referred as both :ref:`CoaddPsf <coaddpsf>` (from :py:class:`lsst.meas.algorithms.CoaddPsf`) and :ref:`StackFit <stackfit>` (after the shear estimation technique where it was developed), and is essentially a form of lazy evaluation.  When a PSF model image is requested at a point, we simply evaluate the PSF models for all of the input images at that point, transform them to the correct coordinate system, and compute the weighted sum on demand.  We typically assume that the PSF is constant over the scale of a single astronomical object, and hence this reduces the number of PSF model evaluations from the number of pixels to the number of detected objects.  When an object lies on a boundary or a region with masked pixels, the true PSF is discontinuous and the constant-PSF assumption is not valid.  At present, we simply flag objects for which this is true, but this may not work when the number of input images is large; in this regime the number of border and masked regions increases, though the severity of the discontinuities decreases as well.
+
+Our current approach for coadd uncertainty propagation is to compute and store only the variance.  We will likely expand this in the future to storing some approximation to the covariance (e.g. by modeling it as constant within regions where the number of input images is constant).
+
+Direct coadds are lossy, requiring some trade-off between image quality (PSF size) and depth (SNR).  This can be easily seen from :eq:`eq:coaddpsf`: including an image with a PSF larger than the current weighted mean PSF always increases the size of the final PSF, regardless of the depth of the new image.
+
+
+PSF-Matched Coadds
+------------------
+
+In PSF-matched coaddition, input images are convolved by a kernel that matches their PSF to a predefined constant PSF before they are combined.  If :math:`\phi_\mathrm{pm}({\bf r})` is the predefined PSF for the coadd, then the matching kernel :math:`K_i({\bf r}, {\bf s})` is defined such that
+
+.. math::
+  \sum_{\bf u} \! K_i({\bf r}, {\bf u}) \, \phi_i({\bf u}, {\bf s})
+    = \phi_\mathrm{pm}({\bf r}-{\bf s})
+
+Typically :math:`K` is parametrized as a smoothly varying linear combination of basis functions.  The details of fitting it given a target coadd PSF and input image PSF models is beyond the scope of this document.
```

Comment:
Is there a good reference document to point the reader at for the details of how to construct K?